patroni/postgresql.py

```diff
@@ -135,16 +164,19 @@ def initialize(self):
     def delete_trigger_file(self):
         os.path.exists(self.trigger_file) and os.unlink(self.trigger_file)
 
-    def sync_from_leader(self, leader):
-        r = parseurl(leader.conn_url)
-
+    def write_pgpass(self, record, append=False):
         pgpass = 'pgpass'
-        with open(pgpass, 'w') as f:
+        with open(pgpass, 'w' if not append else 'a') as f:
```

Comment:
pgpass is only used either by pg_basebackup or by pg_rewind. There is no need to append we can simply rewrite it every time.